### PR TITLE
Mccalluc/user file browser real data

### PR DESCRIPTION
**Pull request by mccalluc**
I'm trying to take the existing code for the file browser 2, identify just the methods I need, and test them. Tests fail right now: The test does not see the factory it should be testing. I also need to figure out how to run these tests locally.

### CODE REVIEW

**Comment by jkmarx** [comment]
https://github.com/refinery-platform/refinery-platform/wiki/Client-&-UI-development#grunt-tasks
@mccalluc Running local ui tests

**Comment by mccalluc** [comment]
Not sure what this guideline implies for me:
> For consistency, use factories instead of services, since they are so similar.

I have a `services` directory, which contains `factory.js` and `user-file.js`, the latter of which is actually a factory:
```javascript
.factory('userFileService', [...
```
- Should the directory be named something different?
- Should the `factory.js` have a more descriptive name, if everything is a factory?

File: refinery/ui/source/js/user-file-browser/services/factory.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      var userFile = userFileService.query();
+      userFile.$promise.then(function (/* response */) {
+        // TODO: addNodeDetailtoUserFiles();
+      }, function (error) {
+        $log.error(error);
+      });
+      return userFile.$promise;
+    }
+
+    // TODO: I think I'll need this soon-ish.
```
Comment (jkmarx) [inline refinery/ui/source/js/user-file-browser/services/factory.js]: Avoid adding commented code.

File: refinery/ui/source/js/user-file-browser/services/factory.spec.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        spyOn(userFileService, 'query').and.callFake(function () {
+          var deferred = $q.defer();
+          deferred.resolve(userFiles);
+          return {
+            $promise: deferred.promise
+          };
+        });
+
+        spyOn(nodeService, 'query').and.callFake(function () {
+          var deferred = $q.defer();
```
Comment (jkmarx) [inline refinery/ui/source/js/user-file-browser/services/factory.spec.js]: why mock nodeService, if you're not using it

File: refinery/ui/source/js/user-file-browser/services/user-file.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+'use strict';
```
Comment (jkmarx) [inline refinery/ui/source/js/user-file-browser/services/user-file.js]: wrap in iffy, according to style guide

Comment (jkmarx) [inline refinery/ui/source/js/user-file-browser/services/user-file.js]: other styling issues....see common service tool_definitions for reference

Comment (jkmarx) [inline refinery/ui/source/js/user-file-browser/services/user-file.js]: or the tools service

**Comment by mccalluc** [comment]
@jkmarx : I think I've addressed all your comments. I've also gotten rid of the mock data in the grid, and am filling in some of the columns with data from solr. TODOs:
- Missing columns
- Column sorting
- CSV
- Functional facets
Not planning to do infinite scrolling, since it sounds like it was hard, and it'd not necessary for this story: They want to be able to narrow it down to just the ones of interest.

File: refinery/ui/source/js/user-file-browser/services/user-file.spec.js [hunk 4]
```diff
@@ -0,0 +1,197 @@
+'use strict';
```
Comment (jkmarx) [inline refinery/ui/source/js/user-file-browser/services/user-file.spec.js]: iffy

**Comment by mccalluc** [comment]
(Talked with Jennifer and with the iife, the last block is resolved.)

**Pull request merged**